### DISCUSSION TITLE
Import from CSV, If any relational data does not exist in DB then create that and import data in database.

### DISCUSSION THREAD

**Discussion by DevShahidul** [filamentphp/filament#10763] [Help]
I have a Country, State and City model with a relationship, which I defined as hasMany and belongsTo both relationships in models and database schema as foreign id. So, the country has many states, the state has many cities.

so what I want is, to create a city CSV in which I mention all country, state and city names, now what I want if the country or state does not exist in DB then create the country and state. In the cities, the table assigns the country's id and state to their respective column, Someone please guide me on how I can implement this feature.

**Comment by sasori13L** [comment]
Did you happen to solve this issue?

**Reply by DevShahidul** [reply]
Yes, I've solved this. thanks for asking :)

**Reply by sasori13L** [reply]
Can you please share how you solved this problem? I am in the same predicament.

**Comment by Joemires** [comment]
You can achieve this by updating your resolveRecord method, e.g
```php
use App\Models\Product;
 
public function resolveRecord(): ?Product
{
    return Product::firstOrNew([
        'sku' => $this->data['sku'],
    ]);
}
```
